schema/schema.go

```diff
@@ -0,0 +1,416 @@
+// Package schema provides methods to build JSON schema of the collections
+// from user defined models.
+//
+// Collection model is declared using Go struct tags. The "tigrisdb" tag contains field properties.
+// The following properties are recognised:
+//  * primary_key - declares primary key of the collection
+//
+// Composite primary keys defined using optional tag indexes.
+//
+// Special tag value "-" can be specified to skip field persistence. Unexported fields are not persisted too.
+package schema
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/tigrisdata/tigrisdb-client-go/driver"
+)
+
+const (
+	tagName       = "tigrisdb"
+	tagPrimaryKey = "primary_key"
+	tagPII        = "pii"
+	tagEncrypted  = "encrypted"
+	tagSkip       = "-"
+
+	typeInteger = "integer"
+	typeString  = "string"
+	typeBoolean = "boolean"
+	typeNumber  = "number"
+	typeArray   = "array"
+	typeObject  = "object"
+
+	formatInt32    = "int32"
+	formatByte     = "byte"
+	formatDateTime = "date-time"
+)
+
+type Field struct {
+	Name   string   `json:"title,omitempty"`
+	Type   string   `json:"type,omitempty"`
+	Format string   `json:"format,omitempty"`
+	Tags   []string `json:"tags,omitempty"`
```

Comment:
do not support it now, add it in the next release?

Reply:
Yes, it'll be just empty.